File: DC/2022/22/Text Editor - Shift & Caps Lock.py
s = input().split()
f = 0
for i in s:
    if i == 'CAPSLOCK':
        if f:
            f = 0
        else:
            f = 1
    elif len(i) > 5 and i[:5] == "SHIFT":
        x = i.split("+")
        print(x[1].lower(), end='') if f else print(x[1].upper(), end='')
    elif f:
        print(i.upper(), end='')
    else:
        print(i, end='')

File: DC/2022/22/Text Editor - Shift & Caps Lock.java
import java.util.*;
public class Hello {

    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        String s[]=sc.nextLine().split(" ");
        boolean f=false;
        for(String x: s){
            if (x.equals("CAPSLOCK")){
                if(f) f=false;
                else f=true;
            }
            else if (x.charAt(0)=='S' && x.length()>5){
                String t[]=x.split("T");
                if(!f) System.out.print(t[1].substring(1).toUpperCase());
                else System.out.print(t[1].substring(1).toLowerCase());
            }
            else if(f) System.out.print(x.toUpperCase());
            else System.out.print(x);
        }

	}
}
